fhir-persistence-jdbc/src/main/java/com/ibm/fhir/persistence/jdbc/util/JDBCParameterBuildingVisitor.java

```diff
@@ -78,11 +95,14 @@ public boolean visit(String elementName, int elementIndex, Visitable visitable)
     /*====================
      * Primitive Types   *
      ====================*/
-    
+
     @Override
     public boolean visit(java.lang.String elementName, int elementIndex, com.ibm.fhir.model.type.Boolean _boolean) {
         Parameter p = new Parameter();
-        p.setName(name);
+        if (!TOKEN.equals(searchParamType)) {
+            throw invalidComboException(searchParamType, _boolean);
+        }
+        p.setName(searchParamCode);
         p.setValueSystem("http://hl7.org/fhir/special-values");
```

Comment:
further should this be a constant?

Reply:
I actually think it should be available from the Boolean class, perhaps as an annotation.  See https://github.com/IBM/FHIR/issues/353